File: codes/yunyeong/229_majorityelementii.py
class Solution(object):
    def majorityElement(self, nums):
        flag = int(len(nums)/3)
        counter = {}
        arr = []
        for num in nums:
            if num in arr:
                continue
            if num in counter:
                counter[num] += 1
            else:
                counter[num] = 1
            if counter[num] > flag:
                arr.append(num)
        return arr
# O(n) / O(n) - Dict

File: codes/yujo/229_majorityelementii.ts
function majorityElement(nums: number[]): number[] {
  const countMap = new Map<number, number>();
  const result: number[] = [];
  const threshold = Math.floor(nums.length / 3);

  for (const num of nums) {
    countMap.set(num, (countMap.get(num) || 0) + 1);
  }

  for (const [num, count] of countMap.entries()) {
    if (count > threshold) {
      result.push(num);
    }
  }

  return result;
}